### ISSUE TITLE
long path names

### ISSUE THREAD

**Issue by jtktam** [Jnesselr/fusion-360-total-exporter#25]
I checked the output directory and it seems to be making the pathnames extra long and some parts are not needed?

\hub <name>\project <project name>\project name

can we shorten this to
\hub <name>\project <project name>\

**Comment by Jnesselr** [comment]
So there's this cute little... annoyance. I'm like... 95% sure that you can only have one top level root folder and that it MUST be the same as the project name, but I can't find any guarantees that that's the case so I thought it'd be safer to just have the extra path name, even though it annoys the hell out of me.

Can someone find this guarantee somewhere? Because if so, I will straight up make it ignore the top level folder.

**Comment by jtktam** [comment]
i am running into issues with the path name limit.. i have rename my top level project to "a" and the output root path to "d:\a" in hope it would make it work

my next step is to rename the \hub \project to \h\p to "steal" back some characters

**Comment by Jnesselr** [comment]
Are you running on Windows? Have you looked into the enabling of long path names? Also, the script shouldn't totally crash on that, it should just not write out the file. If it's totally crashing, that's a bug I need to fix.

**Comment by jtktam** [comment]
it pops up a dialog and complains about not able to write out the file

let me check the long path names thing

**Comment by jtktam** [comment]
also the resume doesn't seem to work for me on windows, it always restarts

**Comment by Gnindit** [comment]
Also if name of the thing you are drawing has a long name to explain what it is, this name is duplicated in the path
In this case "12 mm belt far away block belts in back"
D:/CADexp\Hub Bulltrix\Project CoreXY new\CoreXY new\Try with rails\12 mm belt far away block belts in back.f3d\12 mm belt far away block belts in back v2\Frame\Bottom plate\Fasteners 1\Screws 2.stl

Maybe could be changed to
D:/CADexp\Hub Bulltrix\Project CoreXY new\CoreXY new\Try with rails\12 mm belt far away block belts in back v2\Frame\Bottom plate\Fasteners 1\Screws 2.stl
And in folder "12 mm belt far away block belts in back" put a folder with the complete big files that are right now in the folder "12 mm belt far away block belts in back.f3d".
Something like this (follow the arrows)
![image](https://user-images.githubusercontent.com/17782127/93721276-1d60ce00-fb8f-11ea-9487-801703e95509.png)



And I also do not understand why the "Project CoreXY new\" is needed at all.
Also maybe "Hub " can be removed?

**Comment by 3d-gussner** [comment]
I just replaced the "Hub" with "H" and removed the "Project xxxx" it works fine.

**Comment by MicroKloon** [comment]
> Are you running on Windows? Have you looked into the enabling of long path names? Also, the script shouldn't totally crash on that, it should just not write out the file. If it's totally crashing, that's a bug I need to fix.

2020-10-01 00:17:56,865 - INFO - Writing iges file "C:/Users/ton/NEXTCLOUD/FusionBackup\Hub Peter Jansen\Project CNC 4\CNC 4\EAO W.630KX.055 Stainless Steel Kiosk Panel UK Keyboard\EAO W.630KX.055 Stainless Steel Kiosk Panel UK Keyboard.f3d\EAO W.630KX.055 Stainless Steel Kiosk Panel UK Keyboard v19.igs"
2020-10-01 00:17:59,738 - ERROR - Failed while working on "EAO W.630KX.055 Stainless Steel Kiosk Panel UK Keyboard"
Traceback (most recent call last):
  File "C:/Users/ton/ProjectArchiver/fusion-360-total-exporter-master/Fusion 360 Total Export.py", line 184, in _write_data_file
    self._write_component(file_folder_path, design.rootComponent)
  File "C:/Users/ton/ProjectArchiver/fusion-360-total-exporter-master/Fusion 360 Total Export.py", line 219, in _write_component
    sub_path = self._take(component_base_path, self._name(component.name))
  File "C:/Users/ton/ProjectArchiver/fusion-360-total-exporter-master/Fusion 360 Total Export.py", line 306, in _take
    os.makedirs(out_path, exist_ok=True)
  File "C:/Users/ton/AppData/Local/Autodesk/webdeploy/production/4143a7d9c7cd58a0f86968fd3809be3e34945ca1/Python/lib\os.py", line 221, in makedirs
    mkdir(name, mode)
FileNotFoundError: [WinError 206] De bestandsnaam of -extensie is te lang: 'C:/Users/ton/NEXTCLOUD/FusionBackup\\Hub Peter Jansen\\Project CNC 4\\CNC 4\\EAO W.630KX.055 Stainless Steel Kiosk Panel UK Keyboard\\EAO W.630KX.055 Stainless Steel Kiosk Panel UK Keyboard.f3d\\EAO W.630KX.055 Stainless Steel Kiosk Panel UK Keyboard v19'
